### PR TITLE
ENH: Use OrderedDict in io.netcdf. Fixes #5537

### LINKED ISSUE #5537: use OrderedDict in io.netcdf
See https://github.com/scipy/scipy/pull/5228#issuecomment-159407562. It's in the `pupynere.py` file in the given CVS link; can be easily synced to scipy.

### PR DESCRIPTION
**Pull request by CallumJHays**
Makes use of orderedDict instead of regular dict in io.netcdf. Scavved
from pupynere.

First commit. Please no hate :-)

### CODE REVIEW

File: scipy/io/netcdf.py [hunk 1]
```diff
@@ -989,4 +994,4 @@ def __setitem__(self, index, data):
 
 
 NetCDFFile = netcdf_file
-NetCDFVariable = netcdf_variable
+NetCDFVariable = netcdf_variable
```
Comment (andyfaff) [inline scipy/io/netcdf.py]: Needs a newline at the end of file.

Comment (CallumJHays) [inline scipy/io/netcdf.py]: Thanks for the feedback, Andy.
I'm new to this stuff so I didn't know about this standard practice until now, thanks for informing me :)
By any means it appears that the issue this pull request meant to solve has already been solved by a past patch without resolving the issue.
Thanks again, though!

Comment (rgommers) [inline scipy/io/netcdf.py]: > By any means it appears that the issue this pull request meant to solve has already been solved by a past patch without resolving the issue.

@CallumJHays I'm not sure exactly what you mean here? Which other patch/PR, and what issue exactly?

**Comment by rgommers** [comment]
Since @CallumJHays repo was deleted, I resubmitted this (with fixes) as gh-6932.

Thanks @CallumJHays

**Pull request closed without merging**